packages/react-instantsearch/examples/autocomplete/src/App-Multi-Index.js

```diff
@@ -1,180 +1,46 @@
-            }}
-            inputProps={{
-              placeholder: 'Type a product',
-              value: this.state.query,
-              onChange: () => {
-              },
-            }}
-            renderSectionTitle={section => section.title}
-            getSectionSuggestions={section => section.hits}
-          />
-        </InstantSearch>
-      </div>
-    );
-  }
-}
-
-class FirstResults extends React.Component {
-  shouldComponentUpdate(nextProps) {
-    return !isEqual(this.props.searchState, nextProps.searchState);
-  }
-
-  render() {
-    return <InstantSearch
-      appId="latency"
-      apiKey="6be0576ff61c053d5f9a3225e2a90f76"
-      indexName="bestbuy"
-      searchState={this.props.searchState}
-      onSearchStateChange={this.props.onSearchStateChange}
-    >
-      <div>
-        <Configure hitsPerPage={3} />
-        <VirtualSearchBox/>
-        <VirtualAutoSuggest indexName="bestbuy" onProps={this.props.onProps}/>
-      </div>
-    </InstantSearch>;
-  }
-}
-
-FirstResults.propTypes = {
-  searchState: React.PropTypes.object.isRequired,
-  onSearchStateChange: React.PropTypes.func.isRequired,
-  onProps: React.PropTypes.func.isRequired,
-};
-
-class SecondResults extends React.Component {
-  shouldComponentUpdate(nextProps) {
-    return !isEqual(this.props.searchState, nextProps.searchState);
-  }
-
-  render() {
-    return <InstantSearch
-      appId="latency"
-      apiKey="6be0576ff61c053d5f9a3225e2a90f76"
-      indexName="ikea"
-      searchState={this.props.searchState}
-      onSearchStateChange={this.props.onSearchStateChange}
-    >
-      <div>
-        <Configure hitsPerPage={3} />
-        <VirtualSearchBox/>
-        <VirtualAutoSuggest onProps={this.props.onProps} indexName="ikea"/>
-      </div>
-    </InstantSearch>;
-  }
-}
-
-SecondResults.propTypes = {
-  searchState: React.PropTypes.object.isRequired,
-  onSearchStateChange: React.PropTypes.func.isRequired,
-  onProps: React.PropTypes.func.isRequired,
-};
-
-const VirtualSearchBox = connectSearchBox(() => null);
-
-const connectAutoComplete = createConnector({
-  displayName: 'AutoComplete',
-
-  getProvidedProps(props, state, search) {
-    const hits = search.results ? search.results.hits : [];
-    return {
-      hits, query: state.query !== undefined ? state.query : '',
-    };
-  },
-});
-
-class AutoComplete extends React.Component {
-  // We trigger an update of the autocomplete if the props changed.
-  componentWillReceiveProps(nextProps) {
-    this.props.onProps(nextProps);
-  }
-
-  render() {
-    return null;
-  }
-}
-
-AutoComplete.propTypes = {
-  onProps: React.PropTypes.func.isRequired,
-};
-
-const VirtualAutoSuggest = connectAutoComplete(AutoComplete);
+const App = () =>
+  <InstantSearch
+    appId="latency"
+    apiKey="6be0576ff61c053d5f9a3225e2a90f76"
+    indexName="ikea">
+    <AutoComplete />
+    <Configure hitsPerPage={1} />
+    <MultiIndexContext indexName="bestbuy">
+      <VirtualAutoComplete />
```

Comment:
MultiIndexContext is not a widget per se. Then, in order to create a derivated helper, we need to have a widget under it. 

Any idea for improving this?